### PR TITLE
fix: Display missing data, change axis

### PR DESCRIPTION
**Pull request by i5x64BIT**
- y-axis changed from automatic to 0-100%
- x-axis changed from utc to days
- axis intervals adjusted
- fill gaps: from 00 to last recorded point
- modify sorting to use momentjs functions (otherwise the gaps aren't handled correctly)
Screenshot 2024-01-26 at 12 05 27

#402

### CODE REVIEW

File: src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx [hunk 1]
```diff
@@ -40,31 +39,84 @@ export default function ArrivalByTimeChart({
...
-  data = useMemo(
-    () => data.filter((item) => !operatorId || item.id === operatorId),
-    [data, operatorId],
-  )
+  data = useMemo(() => {
+    let newData = data.filter((item) => !operatorId || item.id === operatorId)
+    if (newData[0]?.gtfs_route_hour) {
+      const lastId = parseInt(newData[newData.length - 1].id)
+      const fillerArray = []
+      for (let currentId = 0; currentId <= lastId; currentId++) {
```
Comment (NoamGaash) [inline src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:47]: please iterate only valid IDs
```
for(let currentId of newData.map(x=>x.id).filter(unique))
```

**Review by NoamGaash — CHANGES_REQUESTED** [review]
I think we should not fill gaps. We should hide the NaN values instead.
Filling gaps with previous data is showing fake data to the user

**Comment by i5x64BIT** [comment]
It doesn't look too bad in the end, see the changes

**Comment by i5x64BIT** [comment]
Hey, linting fails on a file I didn't touch.
Should I commit the fixes for it here?

**Comment by NoamGaash** [comment]
now it should pass.

**Comment by i5x64BIT** [comment]
I see you requested a a change, we discussed on #402 that leaving the data out of the graph is a valid choice.
Screenshot 2024-02-07 at 20 33 15

**Comment by NoamGaash** [comment]
The graph seems amazing, but I didn't have the time to go through the logic.
By the way, can it be simplified?
Anyway, I'll get to it later, thanks for your understanding

File: src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx [hunk 2]
```diff
@@ -76,7 +76,7 @@ export default function ArrivalByTimeChart({
               const currentHour = moment(lastHour).add(1, 'hours').local().toISOString()
               fillerArray.push({
                 ...entry,
-                percent: 0,
+                percent: null,
```
Comment (i5x64BIT) [inline src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:78]: Percent is set to null, thus doesnt appear on the graph.

**Comment by i5x64BIT** [comment]
> The graph seems amazing, but I didn't have the time to go through the logic. By the way, can it be simplified? Anyway, I'll get to it later, thanks for your understanding

Yhea sure, I'll try to explain the main points:
The why - add structure to all graphs (static x and y size)
The how - 
  1. For each unique ID
  2. Get it's points into `allPoints` array
  // Start
  4. startHour = allPoints[0].hour;
  5. While startHour !== 00:00 add null points
  // During
  6. Go over allPoints, whenever the delta between 2 points is bigger then 1 hour, add null points

**Comment by i5x64BIT** [comment]
> > The graph seems amazing, but I didn't have the time to go through the logic. By the way, can it be simplified? Anyway, I'll get to it later, thanks for your understanding
> 
> Yhea sure, I'll try to explain the main points: The why - add structure to all graphs (static x and y size) The how -
> 
> 1. For each unique ID
> 2. Get it's points into `allPoints` array
>    // Start
> 3. startHour = allPoints[0].hour;
> 4. While startHour !== 00:00 add null points
>    // During
> 5. Go over allPoints, whenever the delta between 2 points is bigger then 1 hour, add null points

From my research Recharts seems to print the data as is, it doesn't allow for holes be default.
Data should be parsed before it passes to Recharts, asking chat confirmed it

**Comment by NoamGaash** [comment]
@i5x64BIT see [the suggested pr](https://github.com/i5x64BIT/open-bus-map-search/pull/2) I made two days ago

**Comment by i5x64BIT** [comment]
The test that fails is the legend wrapper, I commited the test removal in this pr.
My design does not have a legened and it looks much cleaner in my opinion.
image

**Review by NoamGaash — APPROVED** [review]
Great!
Thanks for all of your efforts! It looks amazing

**Pull request merged**